care/facility/api/viewsets/asset.py

```diff
@@ -385,9 +385,37 @@ def operate_assets(self, request, *args, **kwargs):
                     "middleware_hostname": middleware_hostname,
                 }
             )
-            result = asset_class.handle_action(action)
+            asset_class.validate_action(action)
+            result = asset_class.handle_action(
+                action,
+                {
+                    "username": request.user.username,
+                    "asset_id": asset.external_id,
+                },
```

Comment:
```suggestion
                username=request.user.username,
                asset_id=asset.external_id,
```
if you use **kwargs in the functions signature then you can directly pass custom parameters no need to use a dict